### PR TITLE
Feature/gh5 powermanager testsuit spec

### PR DESCRIPTION
**Pull request by bhanucbp**
Power Manager test specification for review

### CODE REVIEW

**Review by Anbukannadhasan — COMMENTED** [review]
Added Review Comments

File: docs/pages/powermanager_tests.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+- `L2` - Level 2 Testing ()
+- `L3` - Level 3 Testing ()
+- `IR` - Infrared
+- `CEC` - Consumer Electronics Control
+- `LAN` - Local Area Network
+- `NA` - Not Applicable
+
+## Introduction
+
+This document provides an overview of the testing requirements for the Power Manager module. It outlines the scope of testing, objectives, testing levels and approaches, specific test requirements, and expected deliverables.
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply docs/pages/powermanager_tests.md:33]: done

File: docs/pages/powermanager_tests.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+### Emulator Requirements - Set And Get Power States
+
+`NA`
+
+### Control Plane Requirements - Set And Get Power States
+
+`NA`
+
+## Set and Get Status of Wakeup Sources
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply docs/pages/powermanager_tests.md:68]: done

File: docs/pages/powermanager_tests.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+|-----------|--|--|--------------------------|
+|Set status of various wakeup sources and retrives status for verification |Y|`NA`|`NA`|
+
+### Test Startup Requirement - Set and Get Status of Wakeup Sources
+
+`NA`
+
+### Emulator Requirements - Set and Get Status of Wakeup Sources
+
+Boot configuration: Wakeup sources supported by the device
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply docs/pages/powermanager_tests.md:80]: done

File: docs/pages/powermanager_tests.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+### Control Plane Requirements - Set and Get Status of Wakeup Sources
+
+`NA`
+
+## Testing Wakeup Source
+
+|Description|L2|L3|Control plane requirements|
+|-----------|--|--|--------------------------|
+|Power manager should accept the supported wakeup soruces and device should wakeup from standby/sleep modes using wakeup sources set by caller|`NA`|Y|Control panel to trigger the wakeup source|
```
[the remark this answers is not in the document]
Comment (bhanucbp) [reply docs/pages/powermanager_tests.md:101]: done

File: docs/pages/powermanager_tests.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+### Control Plane Requirements - Set and Get Status of Wakeup Sources
+
+`NA`
+
+## Testing Wakeup Source
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+|1|[Testing Wakeup Source](#testing-wakeup-source)|Power manager should accept the supported wakeup sources and device should wakeup from standby/sleep modes using wakeup sources set by caller|PLAT_API_SetWakeupSrc, PLAT_DS_SetDeepSleep|`NA`|Y|Control plane to trigger the wakeup source|
```
Comment (Anbukannadhasan) [inline docs/pages/powermanager_tests.md:106]: This test can be moved to the DeepSleep manager.

File: docs/pages/powermanager_tests.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+`NA`
+
+### Emulator Requirements - Test Reset Functionality
+
+`NA`
+
+### Control Plane Requirements - Test Reset Functionality
+
+- Discuss options to verify that the device as rebooted successfuly.
+    - Monitor power usage via control panel
```
Comment (Anbukannadhasan) [inline docs/pages/powermanager_tests.md:137]: Not Control panel, its Control Plane.

File: docs/pages/powermanager_tests.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+### Control Plane Requirements - Testing Wakeup Source
+
+Control plane to trigger the wakeup source supported by platform
+
+## Test Reset Functionality
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+|1|[Test Reset Functionality](#test-reset-functionality)|Power manager should reboot the device|PLAT_Reset|`NA`|Y|`NA`||
```
Comment (Anbukannadhasan) [inline docs/pages/powermanager_tests.md:124]: More description is required; the current description will not help generate the L2 low-level spec for ChatGPT.

File: docs/pages/powermanager_tests.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+### Emulator Requirements - Test Reset Functionality
+
+`NA`
+
+### Control Plane Requirements - Test Reset Functionality
+
+- Discuss options to verify that the device as rebooted successfuly.
+    - Monitor power usage via control panel
+    - Verify that the device has rebooted via connectivity with another device
+    - Verify that the reboot as occurred by reconnecting to the device and calling linux `uptime` command
```
Comment (Anbukannadhasan) [inline docs/pages/powermanager_tests.md:139]: Also, required to mention that "Raft framework enables the test to resume from the point where it was interrupted due to a reboot."

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: docs/pages/powermanager_tests.md (hunk 8 above)
Comment (hgfell683) [reply docs/pages/powermanager_tests.md:139]: Added it in as requested

File: docs/pages/powermanager_tests.md (hunk 7 above)
Comment (hgfell683) [reply docs/pages/powermanager_tests.md:124]: Increased the size of the description--admittedly I'm not entirely sure what else to add.

File: docs/pages/powermanager_tests.md (hunk 6 above)
Comment (hgfell683) [reply docs/pages/powermanager_tests.md:137]: Corrected the terminology

File: docs/pages/powermanager_tests.md (hunk 5 above)
Comment (hgfell683) [reply docs/pages/powermanager_tests.md:106]: Going to discuss where we want this moved to, and how to handle the crossover functionality between deepsleep and power manager

File: docs/pages/powermanager_tests.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
+# Power Manager Test Document
+
+## Version History
+
+|Date(DD/MM/YY)|Comment|Version|
+|--------------|-------|-------|
```
[the remark this answers is not in the document]
Comment (hgfell683) [reply docs/pages/powermanager_tests.md]: Removed as requested

File: docs/pages/powermanager_tests.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+
+## Introduction
+
+This document provides an overview of the testing requirements for the Power Manager module. It outlines the scope of testing, objectives, testing levels and approaches, specific test requirements, control plane emulator requirements and expected deliverables.
+
+Interface of the test is available here: [Power Manager HAL header](https://github.com/rdkcentral/rdk-halif-power_manager/blob/main/include/plat_power.h)
+
+## Module Description
+
+The Power manger `HAL` provides a set of `APIs` to initialize, query and set the power state, and query and set the wake up source.
```
[the remark this answers is not in the document]
Comment (hgfell683) [reply docs/pages/powermanager_tests.md]: Removed, as the information is already there.

**Review by Anbukannadhasan — DISMISSED** [review]
Approved

**Review by hgfell683 — APPROVED** [review]
Ship it

**Review by Anbukannadhasan — APPROVED** [review]
Ship it

**Pull request merged**